### DISCUSSION TITLE
[13일차 - 지원] 메모이제이션에 대해 설명해주세요

### DISCUSSION THREAD

**Discussion by smarfy99** [zzangCS/zzangInterview#45] [프론트 면접 질문]
memoization에 대해 설명해주세요

**Comment by skyhanull** [comment]
메모이제이션이란 한 번 연산한 값을 메모리에 기억해두었다가 값이 필요할 때 연산을 다시 반복하지 않고 메모리에서 이미 연산된 값을 다시 꺼내쓰는 기법입니다.
주로 재귀 함수나 동적 프로그래밍에 이 기법이 많이 이용됩니다.

**Comment by ez1n** [comment]
Memoization
- 동일한 계산을 반복해야 할 때 이전에 계산했던 값들을 메모리에 저장함으로써 반복 수행을 제거하여 실행 속도를 높이는 방법으로 동적 계획법의 핵심이 되는 기술
- 리액트의 useMemo, useCallback
